### ISSUE TITLE
Support for extra UV sets

### ISSUE THREAD

**Issue by zeekindustries** [Unity-Technologies/com.unity.formats.alembic#307]
Hey guys!

We are working in a project where we require an Alembic sequence to have 4 UV sets, but we noticed that contrary to our FBX test asset where we can see the sets, the ABC version will only import the first two.
We'd like to know if that's something that we can expect to have implemented or if there's any workaround that I could try on the meantime.
Thanks a lot!

**Comment by clusty** [comment]
Hey,

Thanks for the message.
It is indeed the case, we support only the first 2 UV sets. Will see if i can do something for the future, no promises
Are you able to bake your extra UVs as a texture, and use them in a custom material ?

**Comment by zeekindustries** [comment]
Hey Vlad,
Thanks a lot for getting back to me about this! I do am aware about UV sets being used for other things, but since we are not using the motion vectors in this case, we thought on migrating some shaders that used assets from FBX that hold multiple UV sets. We could definitely bake the extra UVs on other textures but I guess that will come with some downsides regarding resolution quality. I was trying to take a look and build the plugin myself modifying the C++ side, but I'm a novice when it comes to cMake. I'll give it a go to the UVs as a texture and if not, will dive deeper onto modifying the plugin.

**Comment by zeekindustries** [comment]
Hey Vlad!
From the README.txt on the Source folder it says that i should run a setup.bat, but the file is empty and there's no build.bat either. Any chance those getting pushed along (or the abci.sln) as I can't figure how to build the plugin on my own? :c
Thanks a lot!

**Comment by clusty** [comment]
So very sorry, that file is badly out of date:
all you need to do is run build.cmd (Windows) or build.sh (OSX Linux). This uses VS 2015 on windows (remove the -G if you want to use something newer)

The package will be built inside *com.unity.formats.alembic*.
the relevant code is inside: *Source/abci/Importer/aiPolyMesh.cpp* 
I can provide more info on how to do it if you need it.

**Comment by zeekindustries** [comment]
Hey Vlad!

Thanks a lot for the guidance, it was brilliant and I can confirm that I can just build the plugin again with VS2015. 
I think I'm missing something because when I import the geometry, the 4 uv sets are created but the info is sort of swapped, the imported UV0 corresponds to the UV3, and then imported UV1, 2 and 3 are the same and correspond to UV2.

On the C++ side I edited:
Source/abci/Importer/aiPolyMesh.cpp
Source/abci/Importer/aiPolyMesh.h
Source/abci/Importer/AlembicImporter.h

On the C# side I edited:
AbcAPI.cs
AlembicMesh.cs

I am attaching the files I modified if that helps, or if you have any more info that could help me find my way around I will be immensely grateful, i think it's steps away from working as needed. Thanks a lot!

[ABCforUnity_ExtraUVsets.zip](https://github.com/Unity-Technologies/AlembicForUnity/files/4101267/ABCforUnity_ExtraUVsets.zip)
